app/graphql/client.tsx

```diff
@@ -316,6 +333,67 @@ const MyUpdateSub = ({ children }: PropsWithChildren) => {
   return <LnUpdateHashPaidProvider value={lastHash}>{children}</LnUpdateHashPaidProvider>
 }
 
+const NetworkErrorToast: React.FC<{ networkError: ServerError | undefined }> = ({
+  networkError,
+}) => {
+  const { LL } = useI18nContext()
+  const { logout } = useLogout()
+
+  React.useEffect(() => {
+    if (!networkError) {
+      return
+    }
+
+    if (networkError.statusCode >= 500) {
+      // TODO translation
+      toastShow({
+        message: (translations) => translations.errors.network.server(),
+        currentTranslation: LL,
+      })
+    }
+
+    if (networkError.statusCode >= 400 && networkError.statusCode < 500) {
+      let errorCode = networkError.result?.errors?.[0]?.code
+
+      if (!errorCode) {
+        switch (networkError.statusCode) {
+          case 401:
+            errorCode = "INVALID_AUTHENTICATION"
+            break
+        }
+      }
+
+      switch (errorCode) {
+        case NetworkErrorCode.InvalidAuthentication:
+          toastShow({
+            message: (translations) => translations.common.reauth(),
+            onHide: logout,
+            currentTranslation: LL,
+          })
+          break
+
+        default:
+          // TODO translation
+          toastShow({
+            message: (translations) => translations.errors.network.request(),
+            currentTranslation: LL,
+          })
+          break
+      }
+    }
+
+    if (networkError.message === "Network request failed") {
+      // TODO translation
+      toastShow({
+        message: (translations) => translations.errors.network.connection(),
+        currentTranslation: LL,
+      })
+    }
+  }, [networkError, LL, logout])
```

Comment:
Where does setNetworkError get called to clear the error after a toast has been shown?

Reply:
why would we need to clear it? 

if there are no error (ie: at startup), this rountine exit early
if there is a new instance of the network, then it display an error message via toast. it won't re-enter the function until a new error happens, at which point it would display a new error message via toast.

Reply:
Makes sense